### PR TITLE
fix memory request and capacity conversion from a quantity to an int64

### LINKED ISSUE #118: Add support for Kubernetes memory requests in millibytes
It was discovered in #115 that Escalator doesn't handle pod's having memory requests in millibytes. This will cause the memory requests for all pods in a node group to be inaccurate whilst there is a pod with memory in the millibytes. This has a flow on effect in that memory request metrics are inaccurate, and Escalator may start scaling down unexpectedly if only memory requests are specified.

Millibytes are supported in Kubernetes - https://github.com/kubernetes/kubernetes/issues/28741 for backwards compatible reasons. 

We will need tests around different memory and cpu formats to handle edge cases like this as well.

### PR DESCRIPTION
**Pull request by awprice**
This fixes the conversion of pod memory request and node memory capacity from a quantity to an int64 for use with metrics. This will allow the handling of millivalues (which are backwards supported in Kubernetes) in the prometheus metrics.

Note: we already handle the percentage usage calculation using millivalues (which is good) so Escalator would have already handled millivalues when calculating utilisation.

I've updated the metric descriptions for `escalator_node_group_mem_request` and `escalator_node_group_cpu_capacity` as well, as they are the number of bytes, **not** a milli value.

This PR also includes a unit test for `k8s. PodIsStatic`.

cc @wendorf 

Fixes #118

### CODE REVIEW

**Comment by wendorf** [comment]
Thanks for addressing this so soon! 👍

File: pkg/k8s/util_test.go [hunk 1]
```diff
@@ -21,6 +21,16 @@ func TestPodIsDaemonSet(t *testing.T) {
...
 }
 
+func TestPodIsStatic(t *testing.T) {
+	staticPod := test.BuildTestPod(test.PodOpts{})
+	staticPod.ObjectMeta.Annotations = make(map[string]string)
+	staticPod.ObjectMeta.Annotations["kubernetes.io/config.source"] = "file"
+	pod := test.BuildTestPod(test.PodOpts{})
+
+	assert.True(t, k8s.PodIsStatic(staticPod))
+	assert.False(t, k8s.PodIsDaemonSet(pod))
```
Comment (patrickshan) [inline pkg/k8s/util_test.go]: This should be `k8s.PodIsStatic(pod)`
